code/modules/mob/living/carbon/xenomorph/castes/queen/queen.dm

```diff
@@ -100,8 +99,17 @@
 	playsound(loc, 'sound/voice/alien_queen_died.ogg', 75, 0)
 
 /mob/living/carbon/xenomorph/queen/xeno_death_alert()
-	return
+	// This should only happen during modes with Psy Points, aka Distress.
+	if(!CHECK_BITFIELD(SSticker.mode.flags_round_type, MODE_PSY_POINTS))
+		return
 
+	// Checks if there are any building restrictions so there is no redundant declaration to the hive.
+	if(hive.building_allowed == XENO_BUILDING_UNRESTRICTED)
+		return
+	
+	// Resets building permissions to unrestricted upon queen death.
+	hive.building_allowed = XENO_BUILDING_UNRESTRICTED
+	xeno_message("With the death of the Queen, we are free from any restrictions on building silos or turrets.")
```

Comment:
like tivi said, maybe just show this if this was restricted?

Reply:
It should be only shown when restricted. Did I accidently delete the check?

Reply:
No, i'm just blind